docs/sdk/src/reference_docs/defensive_programming.rs

```diff
@@ -0,0 +1,589 @@
+//! ## Fixed Point Arithmetic
+//!
+//! The following code uses types from [`sp_arithmetic`].
+//!
+//! Fixed point arithmetic solves the aforementioned problems of dealing with the uncertain
+//! nature of floating point numbers. Rather than use a radix point (`0.80`), a type which
+//! _represents_ a floating point number in base 10, i.e., a **fixed point number**, can be used
+//! instead.
+//!
+//! For use cases which operate within the range of `[0, 1]` types that implement
+//! [`PerThing`](sp_arithmetic::PerThing) are used:
+//! - **[`Perbill`](sp_arithmetic::Perbill), parts of a billion**
+#![doc = docify::embed!("./src/reference_docs/defensive_programming.rs", perbill_example)]
+//! - **[`Percent`](sp_arithmetic::Percent), parts of a hundred**
+#![doc = docify::embed!("./src/reference_docs/defensive_programming.rs", percent_example)]
+//!
+//! Note that `190 / 400 = 0.475`, and that `Percent` represents it as a _rounded down_, fixed point
+//! number (`47`). Unlike primitive types, types that implement
+//! [`PerThing`](sp_arithmetic::PerThing) will also not overflow, and are therefore safe to use.
+//! They adopt the same behavior that a saturated calculation would provide, meaning that if one is
+//! to go over "100%", it wouldn't overflow, but simply stop at the upper or lower bound.
+//!
+//! For use cases which require precision beyond the range of `[0, 1]`, there are a number of other
+//! fixed-point types to use:
+//!
+//! - [`FixedU64`](sp_arithmetic::FixedU64) and [`FixedI64`](sp_arithmetic::FixedI64)
+//! - [`FixedI128`](sp_arithmetic::FixedU128) and [`FixedU128`](sp_arithmetic::FixedI128)
+//!
+//! Similar to types that implement [`PerThing`](sp_arithmetic::PerThing), these are also
+//! fixed-point types, however, they are able to represent larger fractions:
+#![doc = docify::embed!("./src/reference_docs/defensive_programming.rs", fixed_u64)]
+//!
+//! Let's now explore these types in practice, and how they may be used with pallets to perform
+//! safer calculations in the runtime.
+//!
+//! ### 'PerThing' In Practice
+//!
+//! [`sp_arithmetic`] contains a trait called [`PerThing`](sp_arithmetic::PerThing), allowing a
+//! custom type to be implemented specifically for fixed point arithmetic. While a number of
+//! fixed-point types are introduced, let's focus on a few specific examples that implement
+//! [`PerThing`](sp_arithmetic::PerThing):
+//!
+//! - [`Percent`](sp_arithmetic::Percent) - parts of one hundred.
+//! - [`Permill`](sp_arithmetic::Permill) - parts of a million.
+//! - [`Perbill`](sp_arithmetic::Perbill) - parts of a billion.
+//!
+//! Each of these can be used to construct and represent ratios within our runtime.
+//! You will find types like [`Perbill`](sp_arithmetic::Perbill) being used often in pallet
+//! development.  [`pallet_referenda`] is a good example of a pallet which makes good use of fixed
+//! point arithmetic to calculate.
```

Comment:
... to calculate what? The sentence makes me think something should come after this